R/get_mol_kegg.R

```diff
@@ -154,3 +167,14 @@ split_to_list <- function(x, max_len = 10) {
   }
   
 }
+
+#' Get mol files for compound_ids by splitting into groups of 10 and calling .dl_mol_kegg
+#' @noRd
+dl_mol_kegg <- function(compound_ids) {
+  #balances compound_ids into groups of less than 10 to meet API guidelines
+  compound_id_list <- split_to_list(compound_ids, max_len = 10)
+  
+  #maps over list, but returns it to a single character vector to simplify wrangling code
+  purrr::map(compound_id_list, .dl_mol_kegg) %>% 
+    purrr::list_c()
```

Comment:
rather than using map here and having a `.dl_mol_kegg()` function, it might be better to make use or `req_perform_sequential()` or even `req_perform_parallel()` to work with a list of requests.

Reply:
eh, nevermind.  Played around with using req_perform_parallel() instead and you still need lapply or purrr::map(), and it's not much simpler.  I will add a progress bar to this `purrr::map()` though.